network/codec/cbor/cbor_behaviour_test.go

```diff
@@ -0,0 +1,66 @@
+package cbor
+
+import (
+	"testing"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/stretchr/testify/assert"
+
+	cborcodec "github.com/onflow/flow-go/model/encoding/cbor"
+)
+
+// The CBOR network codec uses the cbor.ExtraDecErrorUnknownField option, which
+// causes decoding to return an error when decoding a message which contains an
+// extra field, not present in the target (struct into which we are decoding).
+//
+// This test validates this behaviour.
+func TestBehaviour_DecodeExtraField(t *testing.T) {
+	type model1 struct {
+		A int
+	}
+	type model2 struct {
+		A int
+		B int
+	}
+
+	m2 := model2{
+		A: 100,
+		B: 200,
+	}
+	bz, err := cborcodec.EncMode.Marshal(m2)
+	assert.NoError(t, err)
+
+	var m1 model1
+	err = defaultDecMode.Unmarshal(bz, &m1)
+	assert.Error(t, err)
+	target := &cbor.UnknownFieldError{}
+	assert.ErrorAs(t, err, &target)
+}
+
+// The CBOR network codec uses the cbor.ExtraDecErrorUnknownField option, which
+// causes decoding to return an error when decoding a message which contains an
+// extra field, not present in the target (struct into which we are decoding).
+//
+// This test validates that, when decoding a message which omits a field present
+// in the target, no error is returned.
```

Comment:
I think it would be helpful to provide some context on when utilizing this behaviour is helpful:
```suggestion
// This test validates that, when decoding a message which OMITS a field present
// in the target, no error is returned. 
//
// This behaviour is very useful for downwards compatibility: for example if we add
// a new field B to a struct, nodes running the updated software can still decode
// messages emitted by the old software - with the convention that in the decoded 
// message, field B has the zero-value. 
// However, note that the reverse is not true - specifically the old software cannot
// decode the new struct, even if field B has the zero value, as demonstrated by the 
// test [TestBehaviour_DecodeExtraField] above.
```